rest_api/tests/api_test/utils.py

```diff
@@ -0,0 +1,387 @@
+    response = submit_request('{}&wait={}'.format(response['link'], WAIT))
+    return response
+
+def query_rest_api(suffix='', data=None, headers=None):
+    if headers is None:
+        headers = {}
+    url = _get_client_address() + suffix
+    return submit_request(urllib.request.Request(url, data, headers))
+
+def submit_request(request):
+    response = urllib.request.urlopen(request).read().decode('utf-8')
+    return json.loads(response)
+
+def _delete_genesis():
+    folder = '/var/lib/sawtooth'
+    for the_file in os.listdir(folder):
+        file_path = os.path.join(folder, the_file)
+        try:
+            if os.path.isfile(file_path):
+                os.unlink(file_path)
+        except Exception as e:
+            print(e)
+
+
+def _get_node_chain(node_list):
+    chain_list = []
+    for node in node_list:
+        try:
+            result = requests.get(node + "/blocks").json()
+            chain_list.append(result['data'])
+        except:
+            LOGGER.warning("Couldn't connect to %s REST API", node)
+    return chain_list
+    
+def check_for_consensus(chains , block_num):
+    LOGGER.info("Checking Consensus on block number %s" , block_num)
+    blocks = []
+    for chain in chains:
+        if chain is not None:
+            block = chain[-(block_num + 1)]
+            blocks.append(block)
+        else:
+            return False
+    block0 = blocks[0]
+    for block in blocks[1:]:
+        if block0["header_signature"] != block["header_signature"]:
+            LOGGER.error("Validators not in consensus on block %s", block_num)
+            LOGGER.error("BLOCK DUMP: %s", blocks)
+            return False
+        else:
+            LOGGER.info('Validators in Consensus on block number %s' , block_num)
+    return True
+
+
+def _get_node_list():
+    client_address = _get_client_address()
+    node_list = [_make_http_address(peer) for peer in _get_peers_list(client_address)]
+    node_list.append(_get_client_address())
+    return node_list
+        
+
+def _get_peers_list(rest_client, fmt='json'):
+    cmd_output = _run_peer_command(
+        'sawtooth peer list --url {} --format {}'.format(
+            rest_client,
+            fmt))
+
+    if fmt == 'json':
+        parsed = json.loads(cmd_output)
+
+    elif fmt == 'csv':
+        parsed = cmd_output.split(',')
+
+    return set(parsed)
+
+def _get_node_chains(node_list):
+    chain_list = []
+    for node in node_list:
+        try:
+            result = requests.get(node + "/blocks").json()
+            chain_list.append(result['data'])
+        except:
+            LOGGER.warning("Couldn't connect to %s REST API", node)
+    return chain_list
+    
+def check_for_consensus(chains , block_num):
+    LOGGER.info("Checking Consensus on block number %s" , block_num)
+    blocks = []
+    for chain in chains:
+        if chain is not None:
+            block = chain[-(block_num + 1)]
+            blocks.append(block)
+        else:
+            return False

```

Comment:
not sure if this is right way for checking consensus, if all validators have validated the blocks and chain head is the same across validators then consensus was performed successfully.

Reply:
it checks whether the blocks are same for all validators on a particular block number